url7f43a4752f_shalynnho_pokemonfactory_tgz-pJ8-Networking_ServerJ8/src/agent/ConveyorAgent.java

```diff
@@ -0,0 +1,401 @@
+						state = ConveyorState.TRANSFERRING_KIT;
+						mk.KS = KitStatus.PICKED_UP;
+						// print("About to send kit");
+						sendKit(mk);
+						return true;
+					}
+				}
+			}
+
+			synchronized (kitsOnConveyor) {
+				for (MyKit mk : kitsOnConveyor) {
+					// Send the kit if it has reached the "stop" position on the
+					// conveyor where the kit robot can pick it up.
+					if (mk.KS == KitStatus.ARRIVED_AT_PICKUP_LOCATION) {
+						mk.KS = KitStatus.AWAITING_PICKUP;
+						kitrobot.msgKitReadyForPickup();
+						// return true;
+					}
+				}
+			}
+
+			// Place kit onto conveyor and start moving it into the cell if
+			// a new kit was requested by the kit robot
+			if (numKitsToDeliver > 0) {
+				numKitsToDeliver--;
+				prepareKit();
+				return true;
+			}
+
+			// print("Checking kits on outbound conveyor");
+			synchronized (kitsOnOutboundConveyor) {
+				for (MyKit mk : kitsOnOutboundConveyor) {
+					// Place kit onto conveyor and start moving it out of
+					// the cell if the kit robot has dropped off a completed kit
+					if (mk.KS == KitStatus.AWAITING_DELIVERY) {
+						print(mk.toString() + " is awaiting delivery");
+						mk.KS = KitStatus.MOVING_OUT;
+						deliverKit(mk);
+						return true;
+					}
+				}
+			}
+
+			if (kitsOnConveyor.size() == 0) {
+				kitrobot.msgNoKitsLeftOnConveyor();
+				// start = false;
+				// return true;
+			} else {
+				print("KitRobot needs to pick up kit");
+				kitrobot.msgKitReadyForPickup();
+				// print("There are " + kitsOnConveyor.size()
+				// + " kits on my conveyor list");
+				// return true;
+			}
+
+		}
+
+		/*
+		 * Tried all rules and found no actions to fire. Return false to the
+		 * main loop of abstract base class Agent and wait.
+		 */
+		return false;
+	}
+
+	/*
+	 * Actions
+	 */
+
+	/**
+	 * Generate a new kit to move into the kitting cell.
+	 */
+	private void prepareKit() {
+		print("Requesting new kit");
+		Kit k = new Kit(kitConfig);
+		incomingKit = new MyKit(k);
+		// print("Got a permit");
+		kitsOnConveyor.add(incomingKit);
+		if (mockgraphics != null) {
+			mockgraphics.msgBringEmptyKit(k.kitGraphics);
+		}
+		if (conveyorGraphics != null) {
+			print("Asking conveyor graphics to animate a new kit");
+			if (k.kitGraphics == null) {
+				print("kitGraphics null");
+			}
+			conveyorGraphics.msgBringEmptyKit(k.kitGraphics);
+		}
+		try {
+			animation.acquire();
+		} catch (InterruptedException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		// print("Got permit");
+		stateChanged();
+	}
+
+	/**
+	 * Send an empty kit to the kitrobot
+	 * @param k the kit being sent.
+	 */
+	private void sendKit(MyKit mk) {
+		print("Sending kit to kit robot");
+		if (mockgraphics != null) {
+			mockgraphics.msgGiveKitToKitRobot(mk.kit.kitGraphics);
+		}
+		if (conveyorGraphics != null) {
+			conveyorGraphics.msgGiveKitToKitRobot(mk.kit.kitGraphics);
+		}
+		try {
+			animation.acquire();
+		} catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)